JoinColony/colonyCDapp
# Feat/2197 multi sig max time

## Description

This PR handles the case for any user to be able to reject a multi sig motion after a week and adds block-ingestor support for when a multi sig motion executes, but the action fails, it rejects the motion.
This can't be tested yet, it will be rechecked as part of #2189 
[Block ingestor PR](https://github.com/JoinColony/block-ingestor/pull/241)

## Testing

We're gonna be hacking dates hard. You will again need 2 users and 2 browsers.

1. Run the dev environment, install the multi sig extension
2. As `leela` give user `amy` `Owner` permissions for `Take actions via Multi-Sig`
3. As `leela` create a mint tokens motion, go to the other browser's colony homepage and check for the latest action in the activity widget and don't open it yet
![image](https://github.com/JoinColony/colonyCDapp/assets/23449297/ad96c809-6215-4569-9127-315cdb384b1e)

4. Open up your dev console and paste in:
```JS
var oneWeekInMilliseconds = 7 * 24 * 60 * 60 * 1000 + 100;

Date.now = function() {
    return new Date().getTime() + oneWeekInMilliseconds;
};
```
5. Run `curl -X POST --data '{"jsonrpc":"2.0","method":"evm_increaseTime","params":[604900000],"id":1}' localhost:8545` in a terminal
6. This should've fast forwarded your motion for a week, and the console will return a week later for `Date.now()` until refresh.
7. Click `Reject action` from the meatball menu as `amy`
![image](https://github.com/JoinColony/colonyCDapp/assets/23449297/aa370a1b-79ae-4c79-816b-177514fc52b0)
8. The motion should get rejected, the sidebar changing


## Diffs

**New stuff** ✨

* Another check if user can reject a motion directly if a week has passed
* In the block ingestor, if a motion executed btu the action failed, we just mark the multi sig motion as rejected (can't reproduce yet)

**Changes** 🏗

* Removed the `isSuccess` flag from a multiSig object


Resolves  #2197
